pkg/controller/job/job_controller.go

```diff
@@ -1361,6 +1361,14 @@ func jobSuspended(job *batch.Job) bool {
 	return job.Spec.Suspend != nil && *job.Spec.Suspend
 }
 
+// Wait for Pods to terminate
+func waitForPodsToTerminate(job *batch.Job) bool {
+	if job.Spec.PodFailurePolicy == nil {
```

Comment:
I think we should also return false when the JobPodFailurePolicy feature gate is disabled. 

So the scenario is
1. someone enabled the flag
2. created the job with the new field and podFailurePolicy
3. disabled the flag. 

Should we then respect the new field? I think we shouldn't.

Reply:
Is it okay to leave all this work under the same feature flag as PodDisruptionConditions and JobPodFailurePolicy?

Reply:
should be fine to leave just under `JobPodFailurePolicy`